### PR TITLE
[Segment Replication] Fix flaky test org.opensearch.indices.replication.SegmentReplicationSuiteIT

### LINKED ISSUE #9499: [BUG] Test org.opensearch.indices.replication.SegmentReplicationSuiteIT is flaky
Using the same seed does not always fail the test. We need to run the test multiple times to get the failure (On my local, I got it on the 13th retry)

Build where it failed: https://build.ci.opensearch.org/job/gradle-check/23203/

I was able to reproduce with `main`

```
  2> java.lang.IllegalStateException: Some shards are still open after the threadpool terminated. Something is leaking index readers or store references.
        at __randomizedtesting.SeedInfo.seed([CFC3DCBFE313A077]:0)
        at org.opensearch.node.Node.awaitClose(Node.java:1541)
        at org.opensearch.test.InternalTestCluster$NodeAndClient.close(InternalTestCluster.java:1129)
        at org.opensearch.common.util.io.IOUtils.close(IOUtils.java:89)
        at org.opensearch.common.util.io.IOUtils.close(IOUtils.java:131)
        at org.opensearch.common.util.io.IOUtils.close(IOUtils.java:114)
        at org.opensearch.test.InternalTestCluster.close(InternalTestCluster.java:966)
        at org.opensearch.common.util.io.IOUtils.close(IOUtils.java:89)
        at org.opensearch.common.util.io.IOUtils.close(IOUtils.java:131)
        at org.opensearch.common.util.io.IOUtils.close(IOUtils.java:114)
        at org.opensearch.test.OpenSearchIntegTestCase.clearClusters(OpenSearchIntegTestCase.java:576)
        at org.opensearch.test.OpenSearchIntegTestCase.afterClass(OpenSearchIntegTestCase.java:2283)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.base/java.lang.reflect.Method.invoke(Method.java:578)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:901)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
        at java.base/java.lang.Thread.run(Thread.java:1623)
  2> REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.indices.replication.SegmentReplicationSuiteIT" -Dtests.seed=CFC3DCBFE313A077 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=en-US -Dtests.timezone=UTC -Druntime.java=20
```

```
Tests with failures:
 - org.opensearch.indices.replication.SegmentReplicationSuiteIT.testFullRestartDuringReplication
 - org.opensearch.indices.replication.SegmentReplicationSuiteIT.testDropRandomNodeDuringReplication
 - org.opensearch.indices.replication.SegmentReplicationSuiteIT.testDeleteIndexWhileReplicating
 - org.opensearch.indices.replication.SegmentReplicationSuiteIT.testBasicReplication
 - org.opensearch.indices.replication.SegmentReplicationSuiteIT.classMethod
```

### PR DESCRIPTION
**Pull request by Rishikesh1159**
### Description
This PR fixes flaky test org.opensearch.indices.replication.SegmentReplicationSuiteIT by moving index creation out of setup.

### Related Issues
Resolves #9499 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Poojita-Raj** [comment]
This fix seems to deal with failure called out in issue description. How does this fix the corruption issue mentioned here: https://github.com/opensearch-project/OpenSearch/issues/10029#issuecomment-1736619011

**Review by Poojita-Raj — APPROVED** [review]

**Review by kotwanikunal — APPROVED** [review]

**Comment by gbbafna** [comment]
```
* What went wrong:
Execution failed for task ':distribution:bwc:staged:buildBwcLinuxTar'.
> Building 2.11.0 didn't generate expected file /var/jenkins/workspace/gradle-check/search/distribution/bwc/staged/build/bwc/checkout-2.11/distribution/archives/linux-tar/build/distributions/opensearch-min-2.11.0-SNAPSHOT-linux-x64.tar.gz

```

**Comment by mch2** [comment]
@Rishikesh1159 How was createIndex in setup causing failures?

**Comment by Rishikesh1159** [comment]
> @Rishikesh1159 How was createIndex in setup causing failures?

@mch2 I was not able to repro this but from the logs I see errors might be happening because of same index being used across all tests in class. I also added test logging to catch this if it happens next time.

**Comment by mch2** [comment]
> > @Rishikesh1159 How was createIndex in setup causing failures?
> 
> @mch2 I was not able to repro this but from the logs I see errors might be happening because of index not being closed, few errors from logs seems like they are affect of previous test's operation. I don't have conclusive evidence that this would resolve as it is not reproducible so added test logging to catch this if it happens next time.

This doesn't make much sense to me, `@Before` is executed before each test, and with suite scope the indices are all wiped in an `@After` test.

File: server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationSuiteIT.java [hunk 1]
```diff
@@ -13,15 +13,16 @@
 import org.opensearch.common.settings.Settings;
 import org.opensearch.indices.replication.common.ReplicationType;
 import org.opensearch.test.OpenSearchIntegTestCase;
+import org.opensearch.test.junit.annotations.TestLogging;
 import org.junit.Before;
 
 @OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.SUITE, minNumDataNodes = 2)
+@TestLogging(reason = "Getting trace logs from replication package", value = "org.opensearch.indices.replication:TRACE")
```
Comment (peternied) [inline server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationSuiteIT.java:20]: I am happy to merge a change that adds more logging so we can better root cause a source of failure; however, I don't think the others changes should be merged.

Comment (mch2) [reply server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationSuiteIT.java:20]: +1 - @Rishikesh1159 mind making this update and we can add the test logging?

**Pull request closed without merging**